tests/Unit/API/Google/AdsAssetTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\AdsAsset;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\UnitTest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Tools\HelperTrait\GoogleAdsClientTrait;
+use PHPUnit\Framework\MockObject\MockObject;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\AssetFieldType;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\CallToActionType;
+use Google\Ads\GoogleAds\Util\V11\ResourceNames;
+use Exception;
+
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AdsAssetTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google
+ *
+ * @property MockObject|OptionsInterface $options
+ * @property AdsAsset                    $asset
+ */
+class AdsAssetTest extends UnitTest {
+
+	use GoogleAdsClientTrait;
+
+	protected const TEMPORARY_ID  = -6;
+	protected const TEST_ASSET_ID = 6677889911;
+
+	/**
+	 * Runs before each test is executed.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->ads_client_setup();
+
+		$this->options = $this->createMock( OptionsInterface::class );
+		$this->options->method( 'get_ads_id' )->willReturn( $this->ads_id );
+
+		$this->asset = new AdsAsset();
+		$this->asset->set_options_object( $this->options );
+	}
+
+	public function test_convert_asset_text() {
+		$data = [
+			'field_type' => AssetFieldType::HEADLINE,
+			'content'    => 'Test headline',
+			'id'         => self::TEST_ASSET_ID,
+		];
+		$row  = $this->generate_asset_row_mock( $data );
+
+		$expected = [
+			'id'      => self::TEST_ASSET_ID,
+			'content' => $data['content'],
+		];
+
+		$this->assertEquals( $expected, $this->asset->convert_asset( $row ) );
```

Comment:
💅  Minor since it's a test. But potentially we could refactor these as 

`assertAssetTypeConversion( $type, $data)`